src/main/java/net/dv8tion/jda/api/interactions/commands/build/CommandData.java

```diff
@@ -91,6 +169,39 @@ public interface CommandData extends SerializableData
     @Nonnull
     Command.Type getType();
 
+    /**
+     * The raw permission bitfield representing the default Permissions of this command.
+     * <br>This is -1 if no permissions have been set, 0 would mean nobody can access the command.
+     *
+     * @return raw permission bitfield representing the default Permissions of this command.
+     */
+    long getDefaultPermissionsRaw();
+
+    /**
+     * The default {@link Permission Permissions} of this command.
+     * <br>A user will not be able to execute this command if he does not have any of these permissions.
+     *
+     * @return {@link EnumSet} of containing the default Permissions of this command.
+     */
+    @Nonnull
+    default EnumSet<Permission> getDefaultPermissions()
+    {
+        long defaultPermissions = getDefaultPermissionsRaw();
+        if (defaultPermissions == -1)
+            return EnumSet.noneOf(Permission.class);
+        if (defaultPermissions == 0)
+            return Permission.getPermissions(Permission.ALL_PERMISSIONS);
+        return Permission.getPermissions(getDefaultPermissionsRaw());
```

Comment:
This behavior isn't explained well anywhere

Reply:
The API uses `null` to indicate all perms, and 0 for no perms. So this doesn't make any sense. Also, you should use `EnumSet.allOf(...)` instead of `Permission.ALL_PERMISSIONS`.

Reply:
["Setting it to "0" will prohibit anyone in a guild from using the command unless a specific overwrite is configured or the user has admin permissions."](https://discord.com/developers/docs/interactions/application-commands#application-command-permissions-object-using-default-permissions)